### PR TITLE
[MT-5010] Refactor - Avoid editor HOC wrappers

### PR DESCRIPTION
**Pull request by e1himself**
- Drop unnecessary `withDebounce`  functionality in favour of `useOnChange` meaningful-only changes filter
- Inline `withAvailableWidth` HOC into `Editor` component
- Make the editor rely on `forwardRef` instead of `editorRef` workaround property

### CODE REVIEW

**Review by aspirisen — APPROVED** [review]

File: packages/slate-editor/src/modules/editor/useOnChange.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import type { Value } from './types';
+
+type OnChange = (value: Value) => void;
+
+/**
+ * Wrap the onChange handler to notify outer code only on meaningful updates.
+ */
+export function useOnChange(onChange: OnChange): OnChange {
+    const props = useLatest({ onChange });
```
[the remark this answers is not in the document]
Comment (e1himself) [reply packages/slate-editor/src/modules/editor/useOnChange.ts:14]: Is this not how all hooks work?

[the remark this answers is not in the document]
Comment (aspirisen) [reply packages/slate-editor/src/modules/editor/useOnChange.ts:14]: God it 👍

[the remark this answers is not in the document]
Comment (aspirisen) [reply packages/slate-editor/src/modules/editor/useOnChange.ts:14]: hm, but why do we set `props` as deps in `useCallback`, as i understand the point is not to create new function on every render?

[the remark this answers is not in the document]
Comment (e1himself) [reply packages/slate-editor/src/modules/editor/useOnChange.ts:14]: > hm, but why do we set `props` as deps in `useCallback`, as i understand the point is not to create new function on every render?

Yes, the goal is to have a stable function returned.

Technically both `props` and `prev` can be omitted from deps, as they will never change. Should I remove them?

**Pull request merged**